src/Pagination/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FunctionComponent, useState, useCallback } from 'react'
+import PaginationArrow from './PaginationArrow'
+import clsx from 'clsx'
+
+import styles from './pagination.module.scss'
+
+interface Props {
+  /** How many buttons are in the pagination */
+  totalPages: number
+  /** Which page is currently selected */
+  activePage?: number
+  itemsPerPage?: number
+  /** Adds a name to the count if shown */
+  countName?: string
+  /** Shows the count of the items in the pagination list */
+  showItemCount?: boolean
+  onChange?: Function
+  className?: string
+  [rest: string]: unknown // ...rest property
+}
+
+const Pagination: FunctionComponent<Props> = ({
+  activePage = 1,
+  totalPages,
+  itemsPerPage = 6,
+  showItemCount = true,
+  onChange,
+  countName = '',
+  className,
+  ...rest
+}) => {
+  const pages = []
+  const [currentPage, setCurrentPage] = useState(activePage)
+  const [currentItems, setCurrentItems] = useState(itemsPerPage)
+  const indexOfLastPage = activePage * totalPages
+  const totalItems = itemsPerPage * totalPages
+  const indexOfFirstPage = indexOfLastPage - totalPages + 1
+
+  const setNumberOfPage = useCallback(
+    (number) => {
+      onChange
+      setCurrentPage(number)
+      setCurrentItems(number * itemsPerPage)
+    },
+    [onChange, currentPage, currentItems]
+  )
+
+  const setPreviousPage = () => {
+    setCurrentPage(currentPage - 1)
+    setCurrentItems((currentPage - 1) * itemsPerPage)
+  }
+
+  const setNextPage = () => {
+    setCurrentPage(currentPage + 1)
+    setCurrentItems((currentPage + 1) * itemsPerPage)
+  }
+
+  /** Creates an array based of the number 
+   * given to the totalPages prop */
+  for (let i = 0; i < totalPages; i++) {
```

Comment:
instead of executing this logic on every render, you can put it in a useMemo:

```
const pages = useMemo(() => {
  const newPages = []
  for (let i = 0; i < totalPages; i++) {
        newPages.push(i + 1)
  }
}, [totalPages])
```